packages/expo/cli/utils/editor.ts

```diff
@@ -0,0 +1,44 @@
+import spawnAsync from '@expo/spawn-async';
+import editors from 'env-editor';
+
+import * as Log from '../log';
+
+/** Guess what the default editor is and default to VSCode. */
+export function guessEditor(): editors.Editor {
+  try {
+    return editors.defaultEditor();
+  } catch {
+    return editors.getEditor('vscode');
+  }
+}
+
+/** Open a file path in a given editor. */
+export async function openInEditorAsync(path: string, inputEditor?: string): Promise<boolean> {
+  const editor = inputEditor ? editors.getEditor(inputEditor) : guessEditor();
+
+  Log.debug(`Opening ${path} in ${editor?.name} (bin: ${editor?.binary}, id: ${editor?.id})`);
+  if (editor) {
+    try {
+      await spawnAsync(editor.binary, [path]);
+      return true;
+    } catch (error) {
+      Log.debug(
+        `Failed to auto open path in editor (path: ${path}, binary: ${editor.binary}, preferred: ${inputEditor}):`,
+        error
+      );
+    }
+  }
+
+  if (inputEditor) {
+    Log.error(
+      // TODO: Is this still in use?
+      `Could not resolve editor from environment variable $EXPO_EDITOR="${inputEditor}". Trying again with system default.`
```

Comment:
@byCedric should we keep `$EXPO_EDITOR` around or just utilize the built-in `$EDITOR` variable?

Reply:
I don't think we are using `$EXPO_EDITOR` anymore, [`env-editor` is actually using `$EDITOR`](https://github.com/sindresorhus/env-editor/blob/v0.5.0/index.js#L183). If I recall, it was just for people to pick a different editor for Expo-related stuff. But I think just using `EDITOR` is actually better.